src/main/scala/service/event/EventServiceImpl.scala

```diff
@@ -0,0 +1,57 @@
+package service.event
+
+import entity.Event
+import io.circe.parser.parse
+import requests.Response
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class EventServiceImpl()(implicit ex: ExecutionContext) extends EventService {
+
+  private val headers = Seq(("Content-Type" -> "application/json"), ("charset" -> "utf-8"))
+
+  override def getById(eventId: String): Future[Option[Event]] = {
+    val response: Response = requests.get(s"http://localhost:8082/events/${eventId}", readTimeout = 600000, connectTimeout = 6000000, headers = headers)
+    val result = for {
+      responseAsString <- parse(response.text()).toOption
+      event = responseAsString.as[Event]
+    } yield event.getOrElse(Option.empty.get)
+    requests.get
+    Future(result)
+  }
+
+  override def getActiveEventsFromAllPages: Future[Seq[Event]] = {
```

Comment:
Could you please explain how this method work it seams a little bit odd and inefficient maybe we can figure out another approach ?